fpsim/defaults.py

```diff
@@ -97,101 +97,92 @@ def register_location(name, location_ref):
 
     # Composite empowerment attributes
-    State('financial_autonomy',    0, float),
-    State('decision_making', 0, float),
+    ss.FloatArr('financial_autonomy', default=0),
+    ss.FloatArr('decision_making', default=0),
 
     # Empowerment - fertility intent
-    State('fertility_intent', 0, bool),
-    State('categorical_intent', "cannot", "<U6"),
-    State('intent_to_use', 0, bool),            # for women not on contraception, whether she has intent to use contraception
+    ss.State('fertility_intent', default=False),
+    ss.Arr('categorical_intent', dtype="<U6",
+           default="no"),  # default listed as "cannot", but its overridden with "no" during init
+    ss.State('intent_to_use', default=False),
 
     # Partnership information -- states will remain at these values if use_partnership is False
-    State('partnered',    0, bool),
-    State('partnership_age', -1, float),
+    ss.State('partnered', default=False),
+    ss.FloatArr('partnership_age', default=-1),
 
     # Socioeconomic
-    State('urban', 1, bool),
-    State('wealthquintile', 3, int),       # her current wealth quintile, an indicator of the economic status of her household, 1: poorest quintile; 5: wealthiest quintile
+    ss.State('urban', default=True),
+    ss.FloatArr('wealthquintile', default=3), # her current wealth quintile, an indicator of the economic status of her household, 1: poorest quintile; 5: wealthiest quintile
 
     # Education - states will remain at these values if use_education is False
-    State('edu_objective',      0, float),
-    State('edu_attainment',     0, float),
-    State('edu_dropout',        0, bool),
-    State('edu_interrupted',    0, bool),
-    State('edu_completed',      0, bool),
-    State('edu_started',        0, bool),
-
-    State('child_inds',         -1,     int,    ncols=max_parity),
-    State('birth_ages',         np.nan, float,  ncols=max_parity),  # Ages at time of live births
-    State('stillborn_ages',     np.nan, float,  ncols=max_parity),  # Ages at time of stillbirths
-    State('miscarriage_ages',   np.nan, float,  ncols=max_parity),  # Ages at time of miscarriages
-    State('abortion_ages',      np.nan, float,  ncols=max_parity),  #  Ages at time of abortions
-    # State('short_interval_ages', np.nan, float, ncols=max_parity)  # Ages of agents at short birth interval
+    ss.FloatArr('edu_objective', default=0),
+    ss.FloatArr('edu_attainment', default=0),
+    ss.State('edu_dropout', default=False),
+    ss.State('edu_interrupted', default=False),
+    ss.State('edu_completed', default=False),
+    ss.State('edu_started', default=False),
+
+    # Add these states to the people object. They are not tracked by timestep in the way other states are, so they
+    # need to be added manually. Eventually these will become part of a separate module tracking pregnancies and
+    # pregnancy outcomes.
+    fpa.TwoDimensionalArr('birth_ages', ncols=max_parity),  # Ages at time of live births
```

Comment:
I wonder whether all these results could be stored in analyzers. I can't think of how they'd be easily compared to data or modified so perhaps they're not used often